### PR TITLE
Boundary: Delete RoseFooter files (HDS-2805)

### PR DESCRIPTION
**Pull request by KristinLBradley**
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/HDS-2805)

Related merged PR: https://github.com/hashicorp/boundary-ui/pull/1994

## Description

If merged, this PR will delete the `RoseFooter` component related files.

NOTE: Currently related tests are just commented out. They can either be deleted entirely or else updated to test the `Hds::AppFooter` if desired before merging the PR.

:technologist: [Admin preview](https://boundary-ui-git-hds-2805-remove-rose-footer-hashicorp.vercel.app/)

## Checklist:

- [ ] I have added before and after screenshots for UI changes
- [ ] I have added JSON response output for API changes
- [ ] I have added steps to reproduce and test for bug fixes in the description
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: addons/rose/tests/integration/components/rose/footer-test.js
Comment (KristinLBradley) [inline addons/rose/tests/integration/components/rose/footer-test.js:1]: I wasn't sure if these test files should be entirely deleted or else updated to test the newly added `Hds::AppFooter`. Please let me know what you would like to do.

Comment (DhariniJeeva) [reply addons/rose/tests/integration/components/rose/footer-test.js:1]: @KristinLBradley these can be deleted since they are part of the rose footer.

Comment (DhariniJeeva) [reply addons/rose/tests/integration/components/rose/footer-test.js:1]: We can have a new test for the footer integration within [`ui/integration/components`](https://github.com/hashicorp/boundary-ui/tree/main/ui/admin/tests/integration/components) to check if the links are valid and if the component exists

**Review by calcaide — APPROVED** [review]
Thank you for the 🧹  🙌  Regarding @DhariniJeeva comment about adding integration test for the links, JFYI we created [this ticket](https://hashicorp.atlassian.net/browse/ICU-11869) on our backlog, so we do not forget to add them later 😄 .

Appreciate the work a lot!!

**Review by DhariniJeeva — APPROVED** [review]
Looks great! thank you for the work

**Pull request merged**